openmdao/core/driver.py

```diff
@@ -262,6 +265,15 @@ def _setup_driver(self, problem):
                     for d in all_vars[:-1]:
                         mysystem_outputs.update(d)
 
+            # de-deplicate sets by removing from mysystem_outputs
+            #   if it's already in another set
+            # clone mysystem_outputs so we can remove from the set while iterating
+            #   over its values
+            temp_outputs = mysystem_outputs.copy()
```

Comment:
I think you could replace this loop with:   mysystem_outputs = mysystem_outputs.difference(mydesvars, myobjectives, myconstraints)

Reply:
Nifty! Good catch